src/platform/linux/misc.cpp

```diff
@@ -100,22 +100,43 @@ namespace platf {
 
   fs::path
   appdata() {
+    bool found = false;
+    bool migrate_config = true;
     const char *dir;
+    const char *homedir;
+    fs::path config_path;
+
+    // Get the home directory
+    if ((homedir = getenv("HOME")) == nullptr) {
+      homedir = getpwuid(geteuid())->pw_dir;
+    }
 
     // May be set if running under a systemd service with the ConfigurationDirectory= option set.
     if ((dir = getenv("CONFIGURATION_DIRECTORY")) != nullptr) {
-      return fs::path { dir } / "sunshine"sv;
+      found = true;
+      config_path = fs::path(dir) / "sunshine"sv;
     }
     // Otherwise, follow the XDG base directory specification:
     // https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
-    if ((dir = getenv("XDG_CONFIG_HOME")) != nullptr) {
-      return fs::path { dir } / "sunshine"sv;
+    if (!found && (dir = getenv("XDG_CONFIG_HOME")) != nullptr) {
+      found = true;
+      config_path = fs::path(dir) / "sunshine"sv;
+    }
+    // As a last resort, use the home directory
+    if (!found) {
+      migrate_config = false;
+      config_path = fs::path(homedir) / ".config/sunshine"sv;
     }
-    if ((dir = getenv("HOME")) == nullptr) {
-      dir = getpwuid(geteuid())->pw_dir;
+
+    // migrate from the old config location if necessary
+    if (migrate_config && found) {
+      fs::path old_config_path = fs::path(homedir) / ".config/sunshine"sv;
+      if (old_config_path != config_path && fs::exists(old_config_path)) {
+        fs::rename(old_config_path, config_path);
```

Comment:
let's use the non-throwing variant that takes a `std::error_code`, then if we get an error from that, let's return `old_config_path` instead.

So it would look something like (untested:
```
std::error_code ec;
fs::rename(old_config_path, config_path, ec);
if (ec) {
  return old_config_path;
}
```

Reply:
Missed this one before... resolved now.